src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/ListingService.java

```diff
@@ -159,10 +161,24 @@ public ListingData setCourtAddressFromCaseData(CaseData caseData) {
         return listingData;
     }
 
-    public ListingData processListingHearingsRequest(ListingDetails listingDetails, String authToken) {
+    private boolean areCharsForClaimantsRespValid(List<SubmitEvent> submitEvents, List<String> errors) {
```

Comment:
Why have you put this method here rather than in InvalidCharacterCheck?

`public static List<String> checkNamesForInvalidCharacters(List<SubmitEvent> submitEvents, String type) {`

Reply:
Also functions with side-effects are best avoided

Reply:
what does side effects mean here?

Reply:
a function should just return a value and nothing else. if it does anything else then this is a side-effect of calling the function. in your case you're changing the value of one of the parameters

Reply:
refactored.